### ISSUE TITLE
Document the exact platform and dependencies versions in use

### ISSUE THREAD

**Issue by pombredanne** [nexB/scancode-workbench#101]
This should include the versions and exact download URL for each OS.

- The base OS supported: 
  - Win7 64-bit and up (no 32 bits for now)
  - macOS 10.9 64-bit and up
  - Linux 64-bit on Ubuntu 12.04 and up (TBD Debian/Fedora/CentOS support) (no 32 bits support)
- The version of node.js (e.g. 6.5.x ?)
- The version of of Electron (e.g. 1.4.x ?)
- The version of Python (e.g. Python 2.7 ?) needed for dev

And the all set of NPMs (as documented in the npm-shrinkwrap.json)

**Comment by johnmhoran** [comment]
@jdaguil Here’s where we currently stand on the checklist @pombredanne identified when he opened this issue:

  * The supported operating systems are described in the wiki’s [Platform Support](https://github.com/nexB/aboutcode-manager/wiki/Platform-Support) section.
  * The download URLs for Linux, MacOS and Windows are listed on the [Releases](https://github.com/nexB/aboutcode-manager/releases) page.
  * The versions of node.js and Python required for development are described in the wiki's [Building](https://github.com/nexB/aboutcode-manager/wiki/Building) section.

We've not yet addressed two items on the checklist:

  * The version of Electron (we're currently using v1.4.0).
  * The set of NPMs (as documented in the [npm-shrinkwrap.json](https://github.com/nexB/aboutcode-manager/blob/develop/npm-shrinkwrap.json)).

**Comment by johnmhoran** [comment]
@jdaguil In the wiki's [Building](https://github.com/nexB/aboutcode-manager/wiki/Building) section, we also list "npm 3.10.x or later (run `npm install -g npm`)" for Linux and MacOS (no such listing for Windows).  However, given your recent experience with `npm` (and our updating the `npm-shrinkwrap.json` to `npm 5.2.0`), perhaps these items need to be modified.  

As I recall, we encountered a problem using `npm 5.3.0`.  Should we have similar entries for each of the three platforms, perhaps something like "npm 3.10.x or later but no more recent than 5.2.0"?

**Comment by jdaguil** [comment]
@johnmhoran Yes, that sound good. Please update as you see fit. FWIW https://github.com/nexB/aboutcode-manager/issues/16#issuecomment-322358712 has the link to the original `npm` issue. You might want to check if the issue has been resolved for later versions.

**Comment by johnmhoran** [comment]
Thanks @jdaguil , I'll update.  Thanks as well for the reminder about the https://github.com/nexB/aboutcode-manager/issues/16#issuecomment-322358712 link to the `npm` issue -- I'll check and report back here.

**Comment by johnmhoran** [comment]
@jdaguil  After checking the status of the `npm` issue, I think for the time being we should recommend users not use anything above `npm 5.2.0`.

While `npm 5.2.0` apparently worked OK, with the release of `5.3.0`, a number of users reported that `npm prune --production` failed.  This specific issue ([#17781 npm@5.3: npm prune --production fails](https://github.com/npm/npm/issues/17781)) seems to have been fixed with the release of `npm 5.4.0` and the issue was closed on 8/22/17.  However, a new issue ([#18277 npm@5.4: npm prune --production fails with native addon](https://github.com/npm/npm/issues/18277)) was opened with a somewhat similar problem -- apparently related to native addons -- and remains open, with one commenter noting that the issue

```
seems to have to do with the fact that sqlite3 is a native addon 
(and it uses node-pre-gyp to install prebuilt binaries).
```

**Comment by johnmhoran** [comment]
@jdaguil  The wiki's [Building](https://github.com/nexB/aboutcode-manager/wiki/Building) section has been updated re `npm`.

Do we want to say anything about the two not-yet-addressed checklist items (repeated below)?

  * The version of Electron (we're currently using v1.4.0).
  * The set of NPMs (as documented in the [npm-shrinkwrap.json](https://github.com/nexB/aboutcode-manager/blob/develop/npm-shrinkwrap.json)).

**Comment by jdaguil** [comment]
@johnmhoran thanks for updating! @pombredanne where did you want those last two documented?

**Comment by OmkarPh** [comment]
Closed in favor of #620
